bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/activities/AbstractActivityManager.java

```diff
@@ -60,4 +62,16 @@ public void removeActivityManagerListener(IActivityManagerListener activityManag
 			activityManagerListeners.remove(activityManagerListener);
 		}
 	}
+
+	@Override
+	public void dispose() {
+		this.activityManagerListeners.clear();
+		this.activityManagerListeners = null;
+		this.isDisposed = true;
```

Comment:
NIT: no need to use `this`

Reply:
all unnecessary usages of `this` fixed with  ada4186